src/earn/hooks.ts

```diff
@@ -40,3 +62,184 @@ export function useEarnPosition(positionId: string = networkConfig.aaveArbUsdcTo
     return pool
   }, [pools, positionId])
 }
+
+export function useDepositEntrypointInfo({
+  allTokens,
+  pool,
+}: {
+  allTokens: TokenBalances
+  pool: EarnPosition
+}) {
+  const { networkId, tokenId: poolTokenId, dataProps } = pool
+
+  const { swappableFromTokens } = useSwappableTokens()
+  const cashInTokens = useCashInTokens()
+  const isSwapEnabled = useSelector(isAppSwapsEnabledSelector)
+  const userLocation = useSelector(userLocationDataSelector)
+
+  const hasDepositToken = useMemo(() => {
+    return allTokens[dataProps.depositTokenId]?.balance?.gt(0) ?? false
+  }, [pool, allTokens])
+  const hasTokensOnSameNetwork = useMemo(() => {
+    return (
+      isSwapEnabled &&
+      !!swappableFromTokens.find(
+        (tokenInfo) =>
+          tokenInfo.networkId === networkId &&
+          tokenInfo.tokenId !== dataProps.depositTokenId &&
+          tokenInfo.tokenId !== poolTokenId &&
+          tokenInfo.balance?.gt(0)
+      )
+    )
+  }, [pool, isSwapEnabled, swappableFromTokens])
+  const hasTokensOnOtherNetworks = useMemo(() => {
+    return (
+      getFeatureGate(StatsigFeatureGates.ALLOW_CROSS_CHAIN_SWAPS) &&
+      isSwapEnabled &&
+      !!swappableFromTokens.find(
+        (tokenInfo) => tokenInfo.networkId !== networkId && tokenInfo.balance?.gt(0)
+      )
+    )
+  }, [pool, isSwapEnabled, swappableFromTokens])
+  const canCashIn = useMemo(() => {
+    return !!cashInTokens.find((tokenInfo) => tokenInfo.tokenId === dataProps.depositTokenId)
+  }, [pool, cashInTokens])
+
+  const asyncExchanges = useAsync(async () => {
+    try {
+      const availableExchanges = await fetchExchanges(
+        userLocation.countryCodeAlpha2,
+        dataProps.depositTokenId
+      )
+
+      return availableExchanges
+    } catch (error) {
+      return []
+    }
+  }, [])
+  const exchanges = useMemo(() => {
+    return asyncExchanges.result ?? []
+  }, [asyncExchanges.result])
+  return { hasDepositToken, hasTokensOnSameNetwork, hasTokensOnOtherNetworks, canCashIn, exchanges }
+}
+
+export function useAddAction({
```

Comment:
do these `use<>Action` need to be hooks? Seems like you could just make these components, just needs to be wrapped with the `ActionCard` right? This could also live closer to where its used. (e.g, in the same file as the component for the bottom sheet)